apps/dashboard/src/main/java/com/akto/action/ApiCollectionsAction.java

```diff
@@ -224,6 +224,85 @@ public String removeApisFromCustomCollection(){
         return SUCCESS.toUpperCase();
     }
 
+    List<ConditionUtils> conditions;
+
+    public static class ConditionUtils {
+        private CollectionCondition.Type type;
+        private CollectionCondition.Operator operator;
+        private BasicDBObject data;
+
+        public BasicDBObject getData() {
+            return data;
+        }
+
+        public void setData(BasicDBObject data) {
+            this.data = data;
+        }
+
+        public ConditionUtils() {
+        }
+
+        public CollectionCondition.Type getType() {
+            return type;
+        }
+
+        public void setType(CollectionCondition.Type type) {
+            this.type = type;
+        }
+
+        public CollectionCondition.Operator getOperator() {
+            return operator;
+        }
+
+        public void setOperator(CollectionCondition.Operator operator) {
+            this.operator = operator;
+        }
+        
+    }
+
+    private static List<CollectionCondition> generateConditions(List<ConditionUtils> conditions){
+        List<CollectionCondition> ret = new ArrayList<>();
+
+        if (conditions != null) {
+            for (ConditionUtils conditionUtils : conditions) {
+                CollectionCondition condition = CollectionCondition.generateCondition(conditionUtils.getType(),
+                        conditionUtils.getOperator(), conditionUtils.getData());
+                if (condition != null) {
+                    ret.add(condition);
+                }
+            }
+        }
+        return ret;
+    }
+
+    public String createCustomCollection() {
+        if (!isValidApiCollectionName()) {
+            return ERROR.toUpperCase();
+        }
+
+        List<CollectionCondition> conditions = generateConditions(this.conditions);
+
+        ApiCollection apiCollection = new ApiCollection(Context.now(), collectionName, conditions);
+        ApiCollectionsDao.instance.insertOne(apiCollection);
+        
+        ApiCollectionUsers.computeCollectionsForCollectionId(apiCollection.getConditions(), apiCollection.getId());
+        
+        this.apiCollections = new ArrayList<>();
+        this.apiCollections.add(apiCollection);
+
+        return SUCCESS.toUpperCase();
+    }
+
+    int apiCount;
+
+    public String getEndpointsFromConditions(){
+        List<CollectionCondition> conditions = generateConditions(this.conditions);
+
+        apiCount = ApiCollectionUsers.getApisFromConditions(conditions).size();
```

Comment:
Directly issue `count` query instead of issuing a `find` query and doing `.size()` later.